hw2/02-reverse.js

```diff
@@ -1,3 +1,37 @@
 /** Exercise 02 - Reverse **/
 
-// Add your code here
+let main = document.getElementsByTagName('main')[0];
+let button = document.getElementById('reverse');
+button.addEventListener('click', reverseTheNumber);
+
+function reverseTheNumber(event) {
+    let output = document.getElementsByTagName('output')[0];
+    if (output == null) {
+        output = document.createElement('output');
+        output.style.paddingTop = '15px'
+        main.appendChild(output);
+    }
+    let input = document.getElementById('input').value;
+
+    //removing leading zeros, ex 00000000
```

Comment:
Again your code is very nicely written and formatted. Almost all output is correct. No digits gives an error, too many digits gives an error, and too few digits gives an error as expected. However, leading zeroes have been removed. This means that there is an error in the output of 8-digit numbers that have leading zeroes. 

For example, inputting `00001111` 
gives the output: `Error: Please input an 8-digit number`. 
However, 00001111 is a valid 8-digit number, so the correct output should be `00001111--> 11110000`

Everything else looks great here, though!

Reply:
I have removed leading zeros, because having them and making 8 digits, don't make that number a valid 8 didgit number